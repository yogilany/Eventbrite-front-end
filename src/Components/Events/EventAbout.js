import React from 'react'
import { Stack } from 'react-bootstrap/'
import * as MUIcons from '@mui/icons-material'
export const EventAbout = (props) => {
    const MainHeaderStyle = {
        color: "#1e0a3c",

    }
    const AboutParagraphStyle = {
        color: "#6f7287",
        fontSize: "16px"
    }
    return (
        <Stack gap={2} style={{ maxWidth: "50vw" }}>
            <h3 style={MainHeaderStyle}>About this event</h3>
            <Stack className="mb-3" direction="horizontal" gap={5}>
                <div>
                    <MUIcons.AccessTime style={{ color: "#0124e9", border: "1px #f8f7fa solid", borderRadius: "10%", backgroundColor: "#f8f7fa", padding: "0.2rem" }} sx={{ fontSize: 26 }} />
                    <strong>1 day 12 hours</strong>
                </div>
                <div>
                    <MUIcons.WifiRounded style={{ color: "#0124e9", border: "1px #f8f7fa solid", borderRadius: "10%", backgroundColor: "#f8f7fa", padding: "0.2rem" }} sx={{ fontSize: 26 }} />
                    <strong>Mobile eTicket</strong>
                </div>
            </Stack>
            <p style={AboutParagraphStyle}>
                {props.about}
                E7kky to Empower: Beyond the Story! event will take place in Cairo, Egypt on 10th March, 2023 celebrating International Women’s Day through the eyes of Egyptian women; their success, their challenges and their life journey. The event will focus on gathering women and girls together to support each other, learn about each other’s stories as well as networking and valuable discussions, activities and workshops that support women to have the power and knowledge to lead a better life where they can enjoy their rights

                Theme of the event: Women and girls empowerment through inspirational talks from influential women, panel discussions on women empowerment, businesses led by women, opportunities from private and public sectors and NGOs provided to women.

                Flow of the Day: One full day event starts at 9 Am and ends at 9 PM. Most activities are happening in parallel: Panel discussions and talks on the main stage, Workshops in meeting rooms, Service Hub, Exhibition, and Networking in the open area, the event theme is built to be dynamic and interactive. In addition to on-side meetings between stakeholders and finally the Closure will be with an entertainment activity TBD later.
            </p>
        </Stack>

    )
}
export default EventAbout;