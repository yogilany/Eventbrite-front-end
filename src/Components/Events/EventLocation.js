import React from 'react'
import { Container, Row, Col } from 'react-bootstrap/'
import * as Ricons from 'react-icons/md/'
import * as MUIcons from '@mui/icons-material'

export const EventLocation = (props) => {
    const MainHeaderStyle = {
        color: "#1e0a3c",

    }
    const HeaderStyle = {
        color: "#1e0a3c",
        FontFace: "Neue Plak",
        fontWeight: "600"
    }

    const DetailsStyle = {
        fontSize: "40px"
    }

    return (
        <Container>
            <h3 style={MainHeaderStyle}>When and where</h3>
            <Row>

                <Col >
                    <Row>
                        <Col md={1}>
                            <MUIcons.CalendarToday style={{ color: "#0124e9", border: "1px #f8f7fa solid", borderRadius: "10%", backgroundColor: "#f8f7fa", padding: "0.2rem" }} sx={{ fontSize: 24 }} />
                        </Col>
                        <Col>
                            <h6 style={HeaderStyle}>Date and time</h6>
                            <p style={{ color: "#6f7287", fontSize: "small" }}>{props.date}Fri, March 10, 2023,<br></br> 9:00 AM â€“ 9:00 PM EET</p>
                        </Col>
                        <Col md={1} style={{ borderLeft: "1px solid #dbdae3" }}  >
                            <MUIcons.LocationOn style={{ color: "#0124e9", border: "1px #f8f7fa solid", borderRadius: "10%", backgroundColor: "#f8f7fa", padding: "0.2rem" }} sx={{ fontSize: 24 }} />
                        </Col>
                        <Col >
                            <h6 style={HeaderStyle}>Location</h6>
                            <p style={{ color: "#6f7287", fontSize: "small" }}>{props.location}Greek Campus Youssef El-Gendy Ad Dawawin, Cairo Governorate 4280102</p>
                        </Col>
                    </Row>

                </Col>

                <Col>
                </Col>
            </Row>
        </Container>

    )
}


export default EventLocation;
