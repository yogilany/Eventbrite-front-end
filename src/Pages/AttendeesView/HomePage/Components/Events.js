import React, { useEffect, useState } from "react";
import { Container, Row, Col } from "react-bootstrap";
import { CategoriesTaps } from "./Categories";
import CategoriesCards from "./CategoriesCards";
import EventCard from "./EventCard";
import Hr from "../../../../Components/Elements/Hr";
import { MdCalendarMonth } from "react-icons/md";
import { Link } from "react-router-dom";
/**
 * @author Yousef Gilany
 * @description This is the Events section of the HomePage.
 * It shows the events on a specific city depending on the user's location.
 * It can be changed to show events on another city. It shows 8 events at a time.
 * @returns {React.FC}

 * @todo Get data from API and show it.
 * @todo Styling headings
 */
const Events = ({
  location,
  events,
  setFree,
  setOnline,
  setToday,
  loading,
}) => {
  const [moreEvents] = useState([]);

  const [category, setCategory] = useState("All");
  const [isCategoriesShown, setIsCategoriesShown] = useState(true);

  var cards = [];

  for (var i = 1; i <= 8; i++) {
    cards.push(i);
  }

  useEffect(() => {
    // console.log("events: ", events);
  }, [events]);

  return (
    <>
      <CategoriesTaps
        categorySelector={setCategory}
        location={location}
        id="categoriesTaps"
        setFree={setFree}
        setOnline={setOnline}
        setToday={setToday}
      />
      {/* {isCategoriesShown ? (
      ): null} */}
      <CategoriesCards location={location} id="categoriesCards" />
      {/* {isCategoriesShown ? (
      ): null} */}
      <CategoriesCards location={location} id="categoriesCards" />

      {events.length != 0 ? (
        <Container className=" pl-5 pr-5 mb-5" id="eventsByLocationSection">
          <h3 className="heading3">Events in {location}</h3>

          {loading ? (
            <div class="flex items-center justify-center rounded-lg mb-36">
              <div role="status">
                <svg
                  aria-hidden="true"
                  class="w-12 h-12 mr-2 text-gray-200 animate-spin  fill-blue-600"
                  viewBox="0 0 100 101"
                  fill="none"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
                    fill="currentColor"
                  />
                  <path
                    d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
                    fill="currentFill"
                  />
                </svg>
                <span class="sr-only">Loading...</span>
              </div>
            </div>
          ) : (
            <Row className="justify-content-md-right">
              {events
                ? events.slice(0, 12).map((event, index) => (
                    <EventCard
                      key={index}
                      eventTitle={event.basic_info.title}
                      eventDate={event.date_and_time.start_date_time}
                      // eventTime={event.eventTime}
                      eventLocation={event.location.city}
                      eventPrice={event.price}
                      eventOrganizer={event.basic_info.organizer}
                      organizerFollowers={event.organizerFollowers}
                      eventCover={event.image_link}
                      eventID={event.id}
                    />
                  ))
                : "Loading..."}
            </Row>
          )}
          <Link to="/browse-events">
            <button
              className="darkOutlineBtn font-semibold "
              id="seeMoreEventsBtn"
            >
              See More
            </button>
          </Link>
        </Container>
      ) : (
        <Container className="pt-5">
          <Row className="justify-content-md-center">
            <Col md={12} className="flex flex-col items-center pb-20">
              <MdCalendarMonth size={50} color="#39364f" />
              <h3 className="heading3 mb-0">No events in your location</h3>
              <h6 className="heading6">Try another location</h6>
            </Col>
          </Row>
        </Container>
      )}
      {/* {events.length != 0 ? (
        <>
          <Hr />
          <Container className=" pl-5 pr-5 mb-5 mt-5" id="moreEventsSection">
            <Row>
              <Col>
                <h3 className="heading3">More Events</h3>
              </Col>
            </Row>

            <Row className="justify-content-md-right">
              {moreEvents
                ? moreEvents.map((event2, index) => (
                  <EventCard
                    key={index}
                    eventTitle={event2.eventName}
                    eventDate={event2.eventDate}
                    eventTime={event2.eventTime}
                    eventLocation={event2.eventLocation}
                    eventPrice={event2.eventPrice}
                    eventOrganizer={event2.eventOrganizer}
                    organizerFollowers={event2.organizerFollowers}
                  />
                ))
                : "Loading..."}
            </Row>
            <Row className="justify-content-md-center">
              <Col md={4}>
                <Link to="/all-events">
                  <button className="darkOutlineBtn" id="seeMoreEventsBtn">
                    See More
                  </button>
                </Link>
              </Col>
            </Row>
          </Container>
        </>
      ) : null} */}
    </>
  );
};

export default Events;
