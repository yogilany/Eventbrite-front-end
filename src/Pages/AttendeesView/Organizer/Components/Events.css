.wholecard {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  width: 300px;
  height: 350px;
  margin: 3%;
  display: inline-block;
}
.bgimg {
  width: 100%;
  height: 40%;
  background-image: url("../../../../assets/drstrange.jpeg");
}

.wholecard:hover {
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
}

.lovebtn {
  margin-top: 47%;
  margin-left: 85%;
  margin-right: 0;
  border-radius: 60px;
}
.lovebtn:hover {
  text-shadow: 0 1px 0 #b12f27;
  background-color: #f64136;
  border-color: #b12f27;
}

.title {
  margin-left: 20px;
  font-size: 18px;
  font-weight: 600;
  color: black;
}

.date {
  margin-left: 20px;
  font-size: 14px;
  font-weight: 600;
  color: crimson;
}

.location {
  margin-left: 20px;
  font-size: 14px;
  font-weight: 400;
  color: black;
}
.price {
  margin-left: 20px;
  font-size: 14px;
  font-weight: 400;
  color: black;
}
