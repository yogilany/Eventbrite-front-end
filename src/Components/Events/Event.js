import React from 'react'
import { Container, Row, Col, Stack } from 'react-bootstrap/'
import { Layout } from '../../app/layout'
import event_image from '../../assets/event_image.png'
import organizer_avatar from '../../assets/event_organizer_avatar.png'
import EventImage from './EventImage.js'
import EventOrganizerCard from './EventOrganizerCard'
import EventLocation from './EventLocation.js'
import EventAbout from './EventAbout.js'
import { EventShare } from './EventShare'
import { EventAboutOrganizer } from './EventAboutOrganizer'
import { EventTicketCard } from './EventTicketCard'
import * as HIIcons from 'react-icons/hi'

const Event = (props) => {
    const EventNameHeaderStyle = {
        color: "#1e0a3c",
        fontWeight: "bold",
        fontSize: "48px"
    }

    return (
        <Container>
            <Layout>
            </Layout>
            <Container fluid className="pt-3" >
                <Row className="pb-5 ">
                    <Col md={12}>
                        <EventImage img_url={event_image} />
                    </Col>
                </Row>
                <Row className="mb-5">
                    <Stack gap={3}>
                        <div>
                            <Container>
                                <Row>
                                    <Col >
                                        <h3>
                                            {props.event_date} Mar 10
                                        </h3>
                                    </Col>
                                    <Col md={{ offset: 9, span: 1 }}>
                                        <HIIcons.HiOutlineHeart size="2em" />
                                    </Col>
                                    <Col md={{ span: 1 }}>
                                        <HIIcons.HiOutlineUpload size="2em" />
                                    </Col>
                                </Row>
                            </Container>

                            <h1 style={EventNameHeaderStyle}>
                                E7kky to Empower: Beyond the Story!
                            </h1>
                        </div>
                        <p>E7kky to Empower Event to celebrate Egyptian Women Journey and go beyond the story!</p>
                    </Stack>
                    <Col md={6} >
                        <EventOrganizerCard
                            avatar={organizer_avatar}
                            name="E7kky magazine"
                            follower_count="1.5k" />
                    </Col>
                </Row>
                <Row className="mb-5">
                    <EventLocation />
                </Row>
                <Row>
                    <EventAbout />
                </Row>
                <Row className="mb-5 mt-5">
                    <EventShare />
                </Row>
                <Row >
                    <EventAboutOrganizer
                        avatar={organizer_avatar}

                        organizer_name="E7kky magazine"
                        follower_count="1.4k" />
                </Row>
                <Row>

                </Row>

            </Container>
        </Container>

    )
}

export default Event;