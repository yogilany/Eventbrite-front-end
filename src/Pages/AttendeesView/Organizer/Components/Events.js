import React from "react";
import "./Events.css";
function Events(props) {
  return (
    <div className="wholecard">
      <div className="bgimg">
        <div className="likebtn">
          <a
            href="google.com"
            title="Add to Favourites"
            className="btn lovebtn"
          >
            <span>&#x2764;</span>
          </a>
        </div>
        <div className="content">
          <p className="title">{props.title}</p>
          <p className="date">{props.date}</p>
          <p className="location">{props.loc}</p>
          <p className="price">{props.price}</p>
        </div>
      </div>
    </div>
  );
}

export default Events;
