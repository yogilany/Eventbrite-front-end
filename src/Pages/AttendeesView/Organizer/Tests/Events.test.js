import React from "react";
import { shallow } from "enzyme";
import Events from "../Components/Events";
describe("Events component", () => {
  it("renders correctly with props", () => {
    const props = {
      title: "Test Event",
      date: "2022-01-01",
      loc: "Test Location",
      price: "$10",
    };
    const wrapper = shallow(<Events {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
