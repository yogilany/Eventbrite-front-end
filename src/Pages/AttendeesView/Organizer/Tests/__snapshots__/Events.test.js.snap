// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events component renders correctly with props 1`] = `ShallowWrapper {}`;
